Dockerfile

```diff
@@ -5,9 +5,18 @@ FROM python:${PYTHON_VERSION} AS developer
 
 # Add any system dependencies for the developer/build environment here
 RUN apt-get update && apt-get install -y --no-install-recommends \
-    graphviz \
+    graphviz vim \
```

Comment:
Should: Similar to the above discussion: Potentially useful change, but should go in a different PR for cleanliness reasons.

Reply:
I added this as pre-commit was complaining it couldn't find an editor. Both `nano` and `vi` are not in the image for some reason. So I couldn't actually commit anything.

Reply:
Interesting... it just uses vscode as the editor for me?